packages/dd-trace/src/config.js

```diff
@@ -183,9 +185,15 @@ class Config {
       process.env.DD_TRACE_STARTUP_LOGS,
       false
     )
+
+    const inAWSLambda = process.env.AWS_LAMBDA_FUNCTION_NAME !== undefined
+    const inGCPFunction = process.env.K_SERVICE !== undefined || process.env.FUNCTION_NAME !== undefined
```

Comment:
`FUNCTION_NAME` is extremely generic and I would expect it to be set in other situations. Are there more reliable way to ensure that the environment is actually GCP and not some random FaaS?

Reply:
Changed this to check for the following env vars:
`FUNCTION_NAME` && `GCP_PROJECT` (identifies GCP functions with deprecated runtimes)
`K_SERVICE` && `FUNCTION_TARGET` (identifies newer GCP function runtimes)
[unfortunately there's no overlap in env vars set by older/newer gcp runtimes](https://cloud.google.com/functions/docs/configuring/env-var#runtime_environment_variables_set_automatically).